shared/src/main/java/org/consumersunion/stories/server/index/story/FullStoryIndexer.java

```diff
@@ -143,64 +148,36 @@ public void index(SolrServer solrStoryServer, SolrServer solrCollectionServer, S
                 documentIdStmt = null, storyTextStmt = null, documentsSelect = null;
 
         Connection conn = null;
         SupportDataUtils supportDataUtils = null;
-        /**
-         * To debug particular stories, ucomment and adjust the 'narrowing' settings. This will set the indexer at
-         * the particular cycle and the boundaries will match as if you were doing a full index from the first item.
-         *
-         * <h2>Debug Notes</h2>
-         *
-         * If there are issues with particulars Stories, it can be useful to narrow the index to the particular
-         * problem area rather than always re-indexing everything. To do this reliably, it's a good idea to
-         * to keep the index batch  boundaries the same, espicially if there's any chance the issue is caused by a
-         * boundary condition.
-         *
-         *  After loading the appropriate DB snapshot, run the following SQL:
-         *  <pre>
-         *  SELECT COUNT(*) FROM story WHERE id &lt; [crit story id];
-         *  </pre>
-         *  Take that number and int divide by the {@link Indexer#BATCH_SIZE} size (100 at time of writing). This gives
-         *  the cycle count.
-         */
-        // int cycleCount = 463;
-        int cycleCount = 0;
 
-        List<SolrInputDocument> stories;
+        int cycleCount = 0;
+        List<StoryDocument> stories;
         try {
             while (true) {
-                // DEBUG: to narrow the document index, you have to initial the SQL assets on the initial cycle.
-                // if ((cycleCount % 10) == 0 || cycleCount == 463) {
                 if ((cycleCount % 10) == 0) {
                     if (conn != null) {
                         conn.close();
                     }
                     conn = PersistenceUtil.getConnection();
                     supportDataUtils = supportDataUtilsFactory.create(conn);
-                    closeIfNotNull(storyDocStmt, answerSetOnlyStmt, tagsStmt, collectionsStmt,
+                    closeIfNotNull(storiesStmt, answerSetOnlyStmt, tagsStmt, collectionsStmt,
                             selectAddress, answerSetStmt, ownerNameStmt, documentIdStmt);
-                    storyDocStmt = conn.prepareStatement(storyDocSql);
+                    storiesStmt = conn.prepareStatement(storiesSql, TYPE_SCROLL_INSENSITIVE, CONCUR_READ_ONLY);
                     answerSetOnlyStmt = conn.prepareStatement(answerSetOnlySql);
                     tagsStmt = conn.prepareStatement("SELECT value FROM tag WHERE systemEntity=?");
                     collectionsStmt = conn.prepareStatement("SELECT e.id, d.title FROM systemEntity e" +
                             " JOIN collection c ON e.id=c.id" +
                             " JOIN entity n ON e.id=n.id" +
                             " JOIN document d ON d.systemEntity=c.id AND d.systemEntityRelation='BODY'" +
-                            " JOIN document d2 ON d.id=d2.id" +
```

Comment:
I'm guessing this was done like this because the sub-SELECT method is generally much slower. In this case probably not a big issue, but I wonder if the data set gets large? Might want to make a comment re. the alternate method for future reference.